### PR TITLE
Modify tests for Date/DateTime repr changes

### LINKED ISSUE #59: Mirror Dates printing changes
Change introduced in: https://github.com/JuliaLang/julia/pull/30200

### PR DESCRIPTION
**Pull request by fchorney**
In Julia v"1.2.0-DEV.29" Date and DateTime repr changed from
"yyyy-mm-dd" to "Date(yyyy, m, d)", and similarly with DateTime. This
changed that comes out of show, so the tests that check the output
string of Intervals with Dates and DateTimes had to be changed to follow
suit.

Closes #59

### CODE REVIEW

File: test/anchoredinterval.jl [hunk 1]
```diff
@@ -148,141 +148,253 @@ using Intervals: canonicalize
...
+        for test in display_tests
+            interval = test[1]
+            @test string(interval) == test[2]
+            @test sprint(show, interval, context=:compact=>true) == string(interval)
+            if VERSION < v"1.2.0-DEV.29"
+                @test sprint(show, interval) == mod_prefix * test[4]
+            else
+                @test sprint(show, interval) == mod_prefix * test[3]
+            end
+        end
```
Comment (omus) [inline test/anchoredinterval.jl]: We should add tests for `repr` and `sprint(print, ...)` so there behaviour is well defined

File: test/anchoredinterval.jl [hunk 2]
```diff
@@ -148,141 +148,253 @@ using Intervals: canonicalize
...
-            "AnchoredInterval{-5 minutes,DateTime}(2016-08-11T02:00:00, ",
-            "Inclusivity(false, true))",
-        )
+        # Each tuple contains...
+        # (
+        #     Interval,
+        #     String Interpretation,
+        #     show interpretation,
+        #     show interpretation for VERSION < v"1.2.0-Dev.30"
+        # )
```
Comment (omus) [inline test/anchoredinterval.jl]: May want to look into named tuples for this.

File: test/anchoredinterval.jl [hunk 3]
```diff
@@ -148,141 +148,256 @@ using Intervals: canonicalize
...
+        # Types
+        T = Tuple{AnchoredInterval, String, String, String}
+        # Fields
+        F = (:interval, :string, :show, :show_old)
```
[the remark this answers is not in the document]
Comment (fchorney) [reply test/anchoredinterval.jl]: I figured that it's just really verbose to have the type name for each value repeated for every tuple. If we want to do that though we can.

[the remark this answers is not in the document]
Comment (omus) [reply test/anchoredinterval.jl]: Fair enough. How about this then:
```julia
display_tests = NamedTuple{(:interval, :string, :show, :show_old)}.([
    (
        HourEnding(dt),
        "(2016-08-11 HE02]",
        string(
            "AnchoredInterval{-1 hour,DateTime}",
            "(DateTime(2016, 8, 11, 2, 0, 0), Inclusivity(false, true))",
        ),
        string(
            "AnchoredInterval{-1 hour,DateTime}",
            "(2016-08-11T02:00:00, Inclusivity(false, true))",
        ),
    ),
    ...
])
```

[the remark this answers is not in the document]
Comment (fchorney) [reply test/anchoredinterval.jl]: That would work!

**Comment by omus** [comment]
Looks like all that is left is to drop Julia 0.6 from the CI files and from the REQUIRE. Please combine the existing commits before you add that commit.

**Comment by fchorney** [comment]
Would I also change `julia="0.6"` to `julia="0.7"` in `docs/make.jl`?

**Comment by fchorney** [comment]
Not too sure why some of the tests are failing with `LoadError: LoadError: ArgumentError: Unknown time zone "Europe/London"`

**Comment by omus** [comment]
I think you can change the "docs/make.jl" to use 1.0.

> Not too sure why some of the tests are failing with `LoadError: LoadError: ArgumentError: Unknown time zone "Europe/London"`

In the build step the tzdata failed to download.

**Comment by omus** [comment]
https://github.com/JuliaTime/TimeZones.jl/issues/180

**Pull request merged**